clumper/clump.py

```diff
@@ -62,6 +62,42 @@ def read_json(cls, path):
             return Clumper(data)
         return Clumper(json.loads(pathlib.Path(path).read_text()))
 
+    @classmethod
+    def read_jsonl(cls, path, lines=None):
+        """
+        Reads in a jsonl file. You can also specify how many lines you want to read in.
+
+        """
+        data_array = []
+
+        if lines is not None:
+            assert lines >= 0, "Number of lines to read must be non-negative"
+
+        def iterate_over_file(file_handler):
+            nonlocal data_array, lines
+
+            for current_line_nr, json_string in enumerate(file_handler):
+                if lines is not None and current_line_nr == lines:
+                    break
+                json_object = json.loads(json_string)
+                data_array.append(json_object)
+            return data_array
+
+        try:
+            # Case 1 : Open cloud file in stream
+            if path.startswith("https:") or path.startswith("http:"):
+                with urllib.request.urlopen(path) as f:
+                    iterate_over_file(f)
+                    return Clumper(data_array)
+            # Case 2 : Local file
+            with open(path) as f:
+                iterate_over_file(f)
+                return Clumper(data_array)
+
+        except Exception:
+            print("Error occured during parsing jsonl file")
+            raise
```

Comment:
Why not;

```python
raise RunTimeError("Error occured during parsing jsonl file")
```

This is typically a lot safer because then other users of our library can catch the error as well as the error message. Might also be nice to write a test confirming that this error is actually being raised.